### ISSUE TITLE
`json` sometimes fails to parse a string that contains a valid JSON array: Value looks like object, but can't find closing '}' symbol

### ISSUE THREAD

**Issue by OrangeTux** [grafana/loki#14397]
**Describe the bug**
Consider the following line line. It contains 2 attributes "object" and "array". Both attributes hold a
string containing valid JSON. 

```
{ "object": "{\"attribute\": 3}", "array": "[1, 2, 3]" }
```

`json` successfully parses the content of field "object" as json [as demonstrated on LogQL Analyzer](https://grafana.com/docs/loki/latest/query/analyzer/?query=%7C+json+%7C+line_format+%60%7B%7B+.object+%7D%7D%60+%7C+json+&line%5B%5D=%7B+%22object%22%3A+%22%7B%5C%22attribute%5C%22%3A+%5C%22optimus%5C%22%7D%22%2C+%22array%22%3A+%22%5B1%2C+2%2C+3%5D%22+%7D).
However, it fails to parse the content of "array" as JSON. It fails with: 

> Value looks like object, but can't find closing '}' symbol 

Again,[ a demonstration on LogQL analyzer.](https://grafana.com/docs/loki/latest/query/analyzer/?query=%7C+json+%7C+line_format+%60%7B%7B+.array+%7D%7D%60+%7C+json&line%5B%5D=%7B+%22object%22%3A+%22%7B%5C%22attribute%5C%22%3A+%5C%22optimus%5C%22%7D%22%2C+%22array%22%3A+%22%5B1%2C+2%2C+3%5D%22+%7D)

**To Reproduce**
1. Visit[ LogQL Analyzer](https://grafana.com/docs/loki/latest/query/analyzer/?query=%7C+json+%7C+line_format+%60%7B%7B+.array+%7D%7D%60+%7C+json&line%5B%5D=%7B+%22object%22%3A+%22%7B%5C%22attribute%5C%22%3A+%5C%22optimus%5C%22%7D%22%2C+%22array%22%3A+%22%5B1%2C+2%2C+3%5D%22+%7D) to see the issue.

**Expected behavior**
I would expect that parsing of an stringified JSON array works and I can access the elements in the array. E.g., I would expect this would print the value at index 2, which happens to be `3`:

```
{job="analyze"} | json | line_format `{{ .array }}` | json |  line_format `{{ .2 }}`
```

**Environment:**
 -  LogQL Analyzer

![Image](https://github.com/user-attachments/assets/28109c40-9a0f-49a8-b059-588bbc092cf9)

**Comment by JStickler** [comment]
I don't think this is a failure of the JSON parser, but more a limitation of the LogQL Analyzer, which is a JavaScript simulation designed to run basic queries.

**Comment by OrangeTux** [comment]
This query doesn't work in Grafana either, unfortunately.
